pyvista/core/pointset.py

```diff
@@ -689,15 +690,27 @@ def __init__(
             # one cell per point (point cloud case)
             verts = self._make_vertex_cells(self.n_points)
             n_verts = self.n_points
+        elif verts is not None:
+            verts = np.asarray(verts)
+            validate_array(verts.flatten(), must_be_padded=True)
 
         # here we use CellArray since we must specify deep and n_faces, etc.
         if verts is not None:
             self.verts = CellArray(verts, n_verts, deep)  # type: ignore
         if strips is not None:
+            if not isinstance(strips, CellArray):
+                strips = np.asarray(strips)
+                validate_array(strips.flatten(), must_be_padded=True)
             self.strips = CellArray(strips, n_strips, deep)  # type: ignore
         if faces is not None:
+            if not isinstance(faces, CellArray):
+                faces = np.asarray(faces)
+                validate_array(faces.flatten(), must_be_padded=True)
             self.faces = CellArray(faces, n_faces, deep)  # type: ignore
```

Comment:
Here and elsewhere, if `isinstance(faces, CellArray)`, no need to construct a new `CellArray`

Reply:
I agree with that